src/Misc/layoutbin/powershell/Start-AzpTask.ps1

```diff
@@ -0,0 +1,146 @@
+If this isn't full language, fix this first.
+This instance is running from: $($MyInvocation.MyCommand.Path).
+Once Start-AzpTask is running in full language mode, verify that the 
+VstsTaskScript and handler are also in full language mode.
+Each script file used by Start-AzpTask must be signed or whitelisted to receive full language mode.
+The VstsTaskScript requested by this instance is: 
+$name
+The handler being requested by this instance is: 
+$scriptBlockString
+Please note other supporting files may also need to be signed or whitelisted to receive full language mode.
+These can be located using procmon, sysmon or similar tools.
+## Error Causes ##
+Start-AzpTask and the task handler scripts can run in different language modes. 
+Mixing language modes is not supported and will result in errors. 
+These may reflect as dot sourcing errors, or errors about a file not being recognized as 
+a cmdlet, function, script file, or operable program.
+## Advanced Usage ##
+Inline PowerShell from pipeline YML task definition is not signed. It runs in CLM by default.
+Signature blocks do not survive conversion from YML so inline scripts can't be signed.
+If you need full language mode for the task action being run by the handler, 
+you must execute it from disk, with proper signing or whitelisting in place.
+##############################
+
+"@
+
+$VerbosePreference = $DebugOption
+$DebugPreference = $DebugOption
+
+# First we check if the device is in CLM mode by default.
+$clmResults = GetCLM 
+# Now the behavior based on CLM and debug settings.
+# Case 1, device is full language, continue as normal 
+if ( $clmResults -eq "FullLanguage" -and $DebugOption -ceq "Continue") {
+    Write-Verbose "Full Language mode detected, continuing traditional workflow."
+}
+# Case 2, device is constrained language  
+else {
+    # 2a We have a request to display debug data and the device is in CLM
+    if ( $DebugOption -ceq "Continue" ) {
+        $verboseNotesString = $verboseNotes -split "\n" 
+        foreach ( $line in $verboseNotesString ) { 
+            Write-Verbose "$line" 
+        }
+    }
+    else {
+        # 2b We have a request to run in constrained language mode, but verbose is not set.
+        Write-Output "Constrained Language Mode detected, but verbose is not set. "
+        Write-Output "Suggest enabling system diagnostics if you need more information."
+    }
+    Write-Output ""
+}
+
+if ([Console]::InputEncoding -is [Text.UTF8Encoding] -and [Console]::InputEncoding.GetPreamble().Length -ne 0) {
+    [Console]::InputEncoding = New-Object Text.UTF8Encoding $false 
+} 
+
+if (!$PSHOME) { 
+    $null = Get-Item -LiteralPath 'variable:PSHOME' 
+}
+else { 
+    Import-Module -Name ([System.IO.Path]::Combine($PSHOME, 'Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1')) 
+    Import-Module -Name ([System.IO.Path]::Combine($PSHOME, 'Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1')) 
+}
```

Comment:
What does this  `if ... ` do?
Should we fail the task instead, or file a warning?